packages/react-magma-dom/src/components/TreeView/TreeItem.tsx

```diff
@@ -162,13 +164,19 @@ const StyledExpandWrapper = styled.div<{
   height: ${props => props.theme.spaceScale.spacing06};
 `;
 
-const StyledCheckboxWrapper = styled.div<{ theme?: ThemeInterface }>`
+const StyledCheckboxWrapper = styled.div<{
+  theme?: ThemeInterface;
+  hasAdditionalContent?: boolean;
+}>`
   margin-right: ${props => props.theme.spaceScale.spacing03};
   vertical-align: middle;
-  display: inline-flex;
+  display: ${props => (props.hasAdditionalContent ? 'flex' : 'inline-flex')};
+  flex-direction: column;
+  width: ${props => `calc(100% - ${props.theme.spaceScale.spacing03})`};
 `;
 
 const StyledItemWrapper = styled.div<{
+  additionalContent?: React.ReactNode;
```

Comment:
What gets passed here could just be a boolean `hasAdditionalContent?: boolean` since the styled component doesn't need the full react node